manim/mobject/types/vectorized_mobject.py

```diff
@@ -2107,30 +2107,22 @@ def __init__(
         self.num_dashes = num_dashes
         self.positive_space_ratio = positive_space_ratio
         super().__init__(color=color, **kwargs)
-        ps_ratio = self.positive_space_ratio
+        p = self.positive_space_ratio
+        n = self.num_dashes
         if num_dashes > 0:
-            # End points of the unit interval for division
-            alphas = np.linspace(0, 1, num_dashes + 1)
-
-            # This determines the length of each "dash"
-            full_d_alpha = 1.0 / num_dashes
-            partial_d_alpha = full_d_alpha * ps_ratio
-
-            # Shifts the alphas and removes the last dash
-            # to give closed shapes even spacing
+            # Assuming total length is 1
+            dash_len = p / n
```

Comment:
I think this is wrong; this should be to compute the void length and not the dash len. 

As a visualization, look at the second example you attached with the PR : 
![image](https://user-images.githubusercontent.com/36239975/126537769-b5c44a4c-6699-4354-aadc-520fb3168ecd.png)

So, I think you just mixed but dash_len and void_len.

Reply:
I still dont see any problem, as p increases, the dash_len (p/n) is increasing and void_len that is (1-p)/n is decreasing